frontend/apps/crates/components/src/module/_common/edit/entry/dom.rs

```diff
@@ -107,10 +108,19 @@ pub fn render_page_body<Mode, Step, RawData, Base, Main, Sidebar, Header, Footer
         })
     }));
 
-    state.page_body_switcher.load(sig.for_each(|dom| {
-        let body = dominator::body();
-        body.set_inner_html("");
-        dominator::append_dom(&body, dom);
+    state.page_body_switcher.load(sig.for_each(clone!(state => move |dom| {
+        {
```

Comment:
This scope is not strictly necessary, but it helps visually separate the 2 pieces of logic happening here.